config/config.go

```diff
@@ -134,8 +139,37 @@ var (
 
 	// Pelican version
 	PelicanVersion string
+
+	MetadataTimeoutErr *MetadataErr = &MetadataErr{Msg: "Timeout when querying metadata"}
 )
 
+// This function creates a new MetadataError by wrapping the previous error
+func NewMetadataError(err error, msg string) *MetadataErr {
+	return &MetadataErr{
+		Msg: fmt.Sprintf("%s: %v", msg, err),
```

Comment:
I think the err shouldn't be concatenated here.  Otherwise it's done twice later on.  What does it look like in your testing?

Reply:
For my hand testing triggering that error it seems to work okay:
```
TransferError = "Attempt #1: Problem initializing the Pelican Client configuration: Error occured when querying for metadata: Get \"https://osg-htc.org/.well-known/pelican-configuration\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```
Note: it's a timeout error above because I just triggered that error with the timeout instead of the other. No duplication seems to be showing up though.